pkg/controller/common/helpers.go

```diff
@@ -658,3 +661,120 @@ func dedupePasswdUserSSHKeys(passwdUser ign2types.PasswdUser) ign2types.PasswdUs
 
 	return passwdUser
 }
+
+// CalculateConfigFileDiffs compares the files present in two ignition configurations and returns the list of files
+// that are different between them
+func CalculateConfigFileDiffs(oldIgnConfig, newIgnConfig *ign3types.Config) []string {
+
+	// Go through the files and see what is new or different
+	oldFileSet := make(map[string]ign3types.File)
+	for _, f := range oldIgnConfig.Storage.Files {
+		oldFileSet[f.Path] = f
+	}
+	newFileSet := make(map[string]ign3types.File)
+	for _, f := range newIgnConfig.Storage.Files {
+		newFileSet[f.Path] = f
+	}
+	diffFileSet := []string{}
+
+	// First check if any files were removed
+	for path := range oldFileSet {
+		_, ok := newFileSet[path]
+		if !ok {
+			// debug: remove
+			glog.Infof("File diff: %v was deleted", path)
```

Comment:
Do we need this debug log here?

Reply:
suggested alternative at  https://github.com/openshift/machine-config-operator/pull/2802/files#r766787404

Reply:
It was in the original code that I refactored, and it never got taken out. I've seen us look at it  a couple times when debugging. 

I was too chicken to take it out, but yeah it's been in there long enough that we should probably either take the debug comment off it and accept that we're keeping it or remove both the log statement and the comment . :smiley:

Reply:
Also, we lose the added/changed/deleted distinction once we return the list, it's just an array of strings representing "stuff that's different than it was before", but we don't keep "how".

Reply:
right, we will lose why exactly file changes. Let's keep the debug log as it is for now and we can move it in future if it start causing annoyance instead of being helpful.